mlflow/tracking/_tracking_service/client.py

```diff
@@ -261,7 +262,9 @@ def rename_experiment(self, experiment_id, new_name):
         """
         self.store.rename_experiment(experiment_id, new_name)
 
-    def log_metric(self, run_id, key, value, timestamp=None, step=None):
+    def log_metric(
+        self, run_id, key, value, timestamp=None, step=None, synchronous: bool = True
```

Comment:
Can we rename `synchronous` to `async`?

Reply:
@dbczumar - What do you prefer here? From spec I know you said 'synchronous', so just checking again.

Reply:
Yep! Perfect!